### ISSUE TITLE
Repeat "softmax" activations in "mrcnn_class_logits"

### ISSUE THREAD

**Issue by Szy-Young** [matterport/Mask_RCNN#427]
Hi,
Thanks for sharing such a project first!
And I find that in model.py, "fpn_classfier_graph" has already implemented "softmax" activations on mrcnn_class_logits, while in "mrcnn_class_loss_graph" the "tf.nn.sparse_softmax_cross_entropy_with_logits" API expects unscaled logits and internally performs a softmax on logits.
Thus the mrcnn_class_logits seems to be activated twice.

**Comment by Szy-Young** [comment]
Besides, K.sparse_categorical_crossentropy in "rpn_class_loss_graph" also sets "from_logits=True" while the rpn_class_logits has been activated with sotfmax before.

**Comment by RLstat** [comment]
Hi, @Szy-Young , after taking a look at the code, I think the mrcnn_class_logits has not been passed through a softmax activation layer, where the output of the softmax activation is mrcnn_prob. It is the mrcnn_class_logits that has later been passed to the mrcnn_class_loss_graph. So I think it should be fine. Let me know if I am mistaken.

**Comment by waleedka** [comment]
@RLstat 's reply answers the question. Closing the issue.
